lib/utils/authorizeSilently.ts

```diff
@@ -0,0 +1,52 @@
+import { SilentAuthorizationError } from "../errors";
+import { TimeoutID } from "../types";
+
+const SERVER_MESSAGE_TIMEOUT = 10000;
+
+export const authorizeSilently = (uri: string, origin: string) => {
+  return new Promise((resolve, reject) => {
+    let timeoutId: TimeoutID;
+    const iframe = document.createElement("iframe");
+    iframe.style.display = "none";
+    document.body.appendChild(iframe);
+
+    const cleanup = () => {
+      window.removeEventListener("message", onMessageHandler);
+      if (iframe && iframe.parentNode) {
+        iframe.parentNode.removeChild(iframe);
+      }
+      if (timeoutId) {
+        clearTimeout(timeoutId);
+      }
+    };
+
+    const onMessageHandler = (event: MessageEvent) => {
+      if (event.origin !== origin) {
+        return;
+      }
+
+      cleanup();
+
+      if (!event.data.error) {
+        const { code } = event.data.response;
+        return resolve(code);
+      } else {
+        return reject(event.data.error);
```

Comment:
What if `event.data` is `undefined`?

Reply:
Hmm I thought it wasn't necessary to check for it since it is defined in our backend 🤔But I guess it doesn't hurt to double check at SDK's end. What about this commit 3eddfacaf534d923be2e50b502684f6c5636a1cf is it fine like so?

Reply:
it looks great 🙏